clients/da-vinci-client/src/main/java/com/linkedin/davinci/DaVinciBackend.java

```diff
@@ -247,6 +250,19 @@ public DaVinciBackend(
       cacheBackend = cacheConfig
           .map(objectCacheConfig -> new ObjectCacheBackend(clientConfig, objectCacheConfig, schemaRepository));
 
+      String storeName = clientConfig.getStoreName();
+
+      if (getRecordTransformer != null && getRecordTransformer.apply(0) != null) {
+        storeRepository.refreshOneStore(storeName);
+        Store store = storeRepository.getStoreOrThrow(storeName);
+        int version = store.getCurrentVersion();
+
+        DaVinciRecordTransformer clientRecordTransformer = getRecordTransformer.apply(version);
+        recordTransformer = new BlockingDaVinciRecordTransformer(
```

Comment:
This will cache a recordtransformer at 'version' but how does it cycle out as versions change?  This backend object I believe doesn't get recreated on new version creation.

Reply:
Could we add a check [here](https://github.com/linkedin/venice/blob/95bf948550f51b17801204bc27d4de3b26774cb8/clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/AvroGenericDaVinciClient.java#L714) that `getRecordTransformer` isn't null, so that a new `DaVinciBackend` is created every time `initBackend` is called?

Reply:
I don't think we should do that, the code is intended to have DavinciBackend as a singleton object.

I think maybe you should take a look at VersionBackend instead.